PittAPI/course.py

```diff
@@ -16,175 +16,103 @@
-            )
-
-    if not course_details:
-        raise ValueError("The TERM or SUBJECT is invalid")
-
-    return course_details
-
-
-def get_courses_by_req(term, req):
-    """
-    :returns: a list of dictionaries containing the data for all SUBJECT classes in TERM
-
-    :param: term: String, term number
-    :param: req: String, requirement abbreviation
-    """
-
-    req = req.upper()
-
-    url = 'http://www.courses.as.pitt.edu/results-genedreqa.asp?REQ={}&TERM={}'.format(req, term)
+URL = 'http://www.courses.as.pitt.edu/'
+
+CODES = [
+    'ADMPS', 'AFRCNA', 'ANTH', 'ARABIC', 'ASL', 'ARTSC', 'ASTRON', 'BIOETH', 'BIOSC', 'CHEM', 'CHLIT', 'CHIN', 'CLASS',
+    'COMMRC', 'CS', 'EAS', 'ECON', 'ENGCMP', 'ENGFLM', 'ENGLIT', 'ENGWRT', 'FP', 'FR', 'FTDA', 'FTDB', 'FTDC', 'GEOL',
+    'GER', 'GREEK', 'GREEKM',  'HINDI', 'HIST', 'HPS', 'HAA', 'ISSP', 'IRISH', 'ITAL', 'JPNSE', 'JS', 'KOREAN', 'LATIN',
+    'LCTL', 'LING', 'MATH', 'MUSIC', 'NROSCI', 'PERS', 'PHIL', 'PEDC', 'PHYS', 'POLISH', 'PS', 'PORT', 'PSY', 'QUECH',
+    'REL', 'RELGST', 'RUSS', 'SERCRO', 'SLAV', 'SLOVAK', 'SOC', 'SPAN', 'STAT', 'SA', 'SWAHIL', 'SWE', 'THEA', 'TURKSH',
+    'UKRAIN', 'VIET', 'BUSACC', 'BUSECN', 'BUSENV', 'BUSFIN', 'BUSHRM', 'BUSBIS', 'BUSMIS', 'BUSMKT', 'BUSORG',
+    'BUSQOM', 'BUSERV', 'BUSSPP', 'BUSSCM', 'ADMJ', 'BUSERV', 'CDACCT', 'CGS', 'LDRSHP', 'LEGLST', 'NPHS', 'PUBSRV',
+    'AFROTC', 'INFSCI', 'MILS', 'BIOENG', 'CEE', 'CHE', 'COE', 'COEE', 'ECE', 'EE', 'ENGR', 'ENGRPH', 'ENRES', 'FTDH',
+    'IE', 'ME', 'MEMS', 'MSE', 'MSEP', 'PETE', 'PWEA', 'WWW', 'HYBRID', 'UHC', 'BCCC']
+REQUIREMENTS = ['G', 'W', 'Q', 'LIT', 'MA', 'EX', 'PH', 'SS', 'HS', 'NS', 'L', 'IF', 'IFN', 'I', 'A']
+PROGRAMS = ['CLST', 'ENV', 'FILMST', 'MRST',  'URBNST', 'SELF', 'GSWS']
+DAY_PROGRAM, SAT_PROGRAM = 'CGSDAY', 'CGSSAT'
+
+# TODO(azharichenko): Create function to fetch this data directly from the course website to make it consistent.
+TERMS = ['2171', '2174', '2177']
+
+
+def get_courses(term, code):
+    """Returns a list of dictionaries containing all courses queried from code."""
+    col_headers, courses = _retrieve_courses_from_url(URL + _get_subject_query(code, term))
+    return [_extract_course_data(col_headers, course) for course in courses]
+
+
+def _get_subject_query(code, term):
+    """Builds query based on code entered."""
+    code, term = code.upper(), _validate_term(term)
+    if code in CODES:
+        return 'results-subja.asp?TERM={}&SUBJ={}'.format(term, code)
+    elif code in PROGRAMS:
+        return 'results-subjspeciala.asp?TERM={}&SUBJ={}'.format(term, code)
+    elif code in REQUIREMENTS:
+        return 'results-genedreqa.asp?TERM={}&REQ={}'.format(term, code)
+    elif code in DAY_PROGRAM:
+        return '/results-dayCGSa.asp?TERM={}'.format(term)
+    elif code in SAT_PROGRAM:
+        return '/results-satCGSa.asp?TERM={}'.format(term)
+    raise ValueError("Invalid subject")
+
+
+def _validate_term(term, *, valid_terms=TERMS):
```

Comment:
Is it necessary to have `valid_terms` as a kwarg? Since we know that `TERMS` is the only possible set of possible valid terms, why have this?